### ISSUE TITLE
Suggestion: roi.ROI(x1, y1...) argument change

### ISSUE THREAD

**Issue by thomasaarholt** [hyperspy/hyperspy#1227]
Currently ROIs take positional arguments like `line = hs.roi.Line2DROI(x1,y1,x2, y2,linewidth)`.
The issue I have with this is that in order to access the individual coordinates later on (say, after *changing* the coordinates using `line.interactive`), one must do `line.x1`, etc.
If one then wants to create a new line (or rectangle) with those coordinates, one must to
```python
line2 = hs.roi.Line2DROI(line.x1, line.y1, line.x2, line.y2, line.linewidth)
```
Might it not be better to have ROIs take a tuple "xycoords" (or similar) as argument, like numpy's `shape` and hyperspy's `.rebin()` do?

I'm not sure if it should include the linewidth argument - I'm still relatively fresh using hyperspy's ROI module - and there may be reasons for the current behaviour that I'm not aware of.
Thoughts?

**Comment by dnjohnstone** [comment]
if you do 
`line = hs.roi.Line2DROI(x1,y1,x2, y2,linewidth)`
Then make some changes interactively,
then just `line` you'll return the updated roi. So you can just copy that roi to make your new one.

I.e. if you do the following you'll get what you want:

`line = hs.roi.Line2DROI(x1,y1,x2, y2,linewidth)`
make changes with `line.interactive`
line2 = line
Not sure what you mean by "should include linewidth argument"? - the linewidth argument results in integrating over a rectangle of that width when you get your line profile out from the roi. This is essentially the same as DM for example.

**Comment by thomasaarholt** [comment]
You're right - I think I was overthinking it. I still think it would be nice to extract the coordinates in a faster way (in case one wants to save them for later, say, after extracting a line profile that's going into a paper/thesis), but it isn't a must.

**Comment by francisco-dlp** [comment]
Currently the `__repr__` method returns a code to reproduce the current ROI and its state. Could you make a more explicit proposal on how to improve upon this?

**Comment by ericpre** [comment]
On the same topic, from @sem-geologist in https://github.com/hyperspy/hyperspy/issues/2059

> I want address few issues here connected with different roi arguments and ROI objects.
> 
> 1. roi arguments in hyperspy functions lacks of consistency, once it expect top bottom and the left right, while in other functions it asks for left right and then top bottom (i.e. crop_image, align2D).
> 2. However, it would be nice if for roi argument in functions, the real interactive rectangular ROI object could be used, as alternative to tuple, or rectangular ROI object would have method to return unified tuple for roi (useful in the functions requiring roi).
> 
> ## currently found bugs:
> * `estimate_shift2D` function have wrong docstring describing the tuple for roi, which if following it results in wrong roi (which leads to poor performance of alignment), or fails the function as roi gets out of image bounds.
> * Line2DROI().interactive() is broken

**Comment by thomasaarholt** [comment]
@sem-geologist could you write out how you would expect your point #2 to work? Both in terms of the "real interactive ROI" and the "method to return unified tuple"? I'm happy to implement it.

**Comment by thomasaarholt** [comment]
@sem-geologist which aspect of Line2DROI is broken? It seems to work fine here?
